onadata/libs/renderers/renderers.py

```diff
@@ -244,9 +250,12 @@ def _to_xml(self, xml, data):
 
         elif isinstance(data, dict):
             for (key, value) in iteritems(data):
-                xml.startElement(key, {})
-                self._to_xml(xml, value)
-                xml.endElement(key)
+                if key not in FORMLIST_MANDATORY_FIELDS and value is None:
+                    pass
```

Comment:
Could we use the `continue` keyword and avoid the else block? I think it seems cleaner that way.